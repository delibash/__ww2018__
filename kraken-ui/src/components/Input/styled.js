import styled from 'styled-components';
import { colors } from './../../global/constants';

const StyledLabel = styled.label`
  position: relative;
`;

const StyledInput = styled.input`
  min-height: 3.8rem;
  padding: .2rem .8rem;
  border: 0;
  border-radius: .3rem;
  box-shadow: ${colors.BOX_SHADOW_INSET};
  font-size: 1.3rem;
  width: 100%;
`;

const StyledSearchButton = styled.button`
  position: absolute;
  right: .1rem;
  top: .1rem;
  background: ${props => props.disabled ? colors.LIGHTER_GRAY : colors.BLUE};
  color: ${colors.WHITE};
  border: 0 none;
  min-height: 3.6rem;
  border-radius: 0 .3rem .3rem 0;
  cursor: pointer;
`;

export {
  StyledLabel,
  StyledSearchButton,
  StyledInput
};
