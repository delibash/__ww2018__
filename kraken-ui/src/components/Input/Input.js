import React from 'react';
import PropTypes from 'prop-types';
import { StyledInput, StyledSearchButton, StyledLabel } from './styled';
import InlineSVG from 'svg-inline-react';
import { SVGIcons } from './../../global/constants';

const Input = ({type, disabled, ...props}) => (
  type === 'search' ? 
    <StyledLabel htmlFor={props.id}>
      <StyledInput {...props} />
      <StyledSearchButton
        type="submit"
        disabled={disabled}
      >
        <InlineSVG src={SVGIcons.search} />
      </StyledSearchButton>
    </StyledLabel>
    :
    <StyledInput type={type} {...props} />
);

Input.propTypes = {
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  id: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
};

export default Input;
