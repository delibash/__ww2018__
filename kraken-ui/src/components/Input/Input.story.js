import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Input from './Input';

storiesOf('Input', module)
  .add('Text Input', () => (
    <Input
      placeholder="Placeholder"
      type="text"
      onChange={action('input-onchange')}
    />
  ));


