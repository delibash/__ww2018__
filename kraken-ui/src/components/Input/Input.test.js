import React from 'react';
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Input from './Input';

test('Default Button', () => {
  const tree = renderer.create(
    <Input
      type="text"
      onChange={() => {}}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});