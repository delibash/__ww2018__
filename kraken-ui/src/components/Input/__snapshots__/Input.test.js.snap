// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button 1`] = `
.c0 {
  min-height: 3.8rem;
  padding: .2rem .8rem;
  border: 0;
  border-radius: .3rem;
  box-shadow: inset 0 0 0 .1rem rgba(0,0,0,.25);
  font-size: 1.3rem;
  width: 100%;
}

<input
  className="c0"
  onChange={[Function]}
  type="text"
/>
`;
